gobblin-core/src/main/java/org/apache/gobblin/converter/parquet/ParquetGroup.java

```diff
@@ -0,0 +1,247 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.gobblin.converter.parquet;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import parquet.example.data.Group;
+import parquet.example.data.simple.BinaryValue;
+import parquet.example.data.simple.BooleanValue;
+import parquet.example.data.simple.DoubleValue;
+import parquet.example.data.simple.FloatValue;
+import parquet.example.data.simple.Int96Value;
+import parquet.example.data.simple.IntegerValue;
+import parquet.example.data.simple.LongValue;
+import parquet.example.data.simple.NanoTime;
+import parquet.example.data.simple.Primitive;
+import parquet.io.api.Binary;
+import parquet.io.api.RecordConsumer;
+import parquet.schema.GroupType;
+import parquet.schema.PrimitiveType;
+import parquet.schema.Type;
+
+import static parquet.schema.Type.Repetition.REPEATED;
+
+
+/**
+ * Custom Implementation of {@link Group} to support adding {@link Object} of type {@link Primitive} or {@link Group}.
+ * Also provides methods to add {@link Primitive} and {@link Group} with {@link String} key if index is not known.
+ * @author tilakpatidar
+ */
+public class ParquetGroup extends Group {
+
+  private final GroupType schema;
+  //data indexed by fieldIndex of the schema
+  private final List<Object>[] data;
+
+  public ParquetGroup(GroupType schema) {
+    this.schema = schema;
+    this.data = new List[schema.getFields().size()];
+
+    for (int i = 0; i < schema.getFieldCount(); ++i) {
+      this.data[i] = new ArrayList();
+    }
+  }
+
+  public String toString() {
+    return this.toString("");
+  }
+
+  public String toString(String indent) {
+    StringBuilder result = new StringBuilder();
+    int i = 0;
+    Iterator<parquet.schema.Type> i$ = this.schema.getFields().iterator();
+
+    while (true) {
```

Comment:
This block from line 71 to 97 has 3 loops. Can we simply the logic as follows?
``` 
StringBuilder result = new StringBuilder();
List<Field> fields = this.schema.getFields();
for (int i = 0; i < fields.length; i++) {
  data[i].forEach(fieldValue -> {
    result.append(indent).append(fields.get(i).getName());
    if (fieldValue == null) {
      result.append(": NULL\n");
    } else if (fieldValue instanceof Group) {
      result.append("\n").append(((ParquetGroup) fieldValue).toString(indent + "  "));
    } else {
      result.append(": ").append(fieldValue.toString()).append("\n");
    }
  });
}
return result.toString();
```

Reply:
👍